File: apina-core/src/test/java/fi/evident/apina/model/type/ApiClassTypeTest.java
package fi.evident.apina.model.type;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.not;
import static org.junit.Assert.assertThat;

public class ApiClassTypeTest {

    @Test
    public void equality() {
        assertThat(new ApiClassType(new ApiTypeName("foo.Bar")), is(new ApiClassType(new ApiTypeName("foo.Bar"))));
        assertThat(new ApiClassType(new ApiTypeName("foo.Bar")), is(not(new ApiClassType(new ApiTypeName("foo.Baz")))));
    }
}

File: apina-core/src/test/kotlin/fi/evident/apina/model/type/ApiClassTypeTest.kt
package fi.evident.apina.model.type

import fi.evident.apina.model.type.ApiType.Class
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class ApiClassTypeTest {

    @Test
    fun equality() {
        assertEquals(Class(ApiTypeName("foo.Bar"), emptyList()), Class(ApiTypeName("foo.Bar"), emptyList()))
        assertNotEquals(Class(ApiTypeName("foo.Bar"), emptyList()), Class(ApiTypeName("foo.Baz"), emptyList()))
    }
}
